File: src/Application Specific/Classes/Entity Collections/Warehouse/Warehouse.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace Warehouses

Public Class Warehouse

    Private _dataTypeValidator As DataTypeValidator
    Public Sub New()
        _dataTypeValidator = New DataTypeValidator
    End Sub

#Region "Class Properties"

    Public Property IgnoreExceptionsOnSetMethods() As Boolean
        Get
            Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
        End Get
        Set(ByVal Value As Boolean)
            Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
        End Set
    End Property

    Public ReadOnly Property Errors() As Hashtable
        Get
            Return _dataTypeValidator.Errors
        End Get
    End Property

    Public ReadOnly Property DTValidator() As DataTypeValidator
        Get
            Return _dataTypeValidator
        End Get
    End Property
    
    Private _exists As Boolean = False
    Public Property Exists() As Boolean
        Get
            Return _exists
        End Get
        Set(ByVal Value As Boolean)
            _exists = Value
        End Set
    End Property

    Private _deleted As Boolean = False
    Public ReadOnly Property Deleted() As Boolean
        Get
            Return _deleted
        End Get
    End Property
    Public WriteOnly Property SetDeleted() As Boolean
        Set(ByVal Value As Boolean)
            _deleted = Value
        End Set
    End Property
    
#End Region

#Region "Warehouse Properties"

      
Private _WarehouseID As Integer = 0
Public Readonly Property WarehouseID() As Integer
	Get 
		Return _WarehouseID
	End Get	
End Property	
Public Writeonly Property SetWarehouseID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_WarehouseID", Value)
	End Set
End Property


Private _Name As String = String.Empty
Public Readonly Property Name() As String
	Get 
		Return _Name
	End Get	
End Property	
Public Writeonly Property SetName() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Name", Value)
	End Set
End Property


Private _Size As String = String.Empty
Public Readonly Property Size() As String
	Get 
		Return _Size
	End Get	
End Property	
Public Writeonly Property SetSize() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Size", Value)
	End Set
End Property


Private _Notes As String = String.Empty
Public Readonly Property Notes() As String
	Get 
		Return _Notes
	End Get	
End Property	
Public Writeonly Property SetNotes() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Notes", Value)
	End Set
End Property


Private _Address1 As String = String.Empty
Public Readonly Property Address1() As String
	Get 
		Return _Address1
	End Get	
End Property	
Public Writeonly Property SetAddress1() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Address1", Value)
	End Set
End Property


Private _Address2 As String = String.Empty
Public Readonly Property Address2() As String
	Get 
		Return _Address2
	End Get	
End Property	
Public Writeonly Property SetAddress2() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Address2", Value)
	End Set
End Property


Private _Address3 As String = String.Empty
Public Readonly Property Address3() As String
	Get 
		Return _Address3
	End Get	
End Property	
Public Writeonly Property SetAddress3() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Address3", Value)
	End Set
End Property


Private _Address4 As String = String.Empty
Public Readonly Property Address4() As String
	Get 
		Return _Address4
	End Get	
End Property	
Public Writeonly Property SetAddress4() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Address4", Value)
	End Set
End Property


Private _Address5 As String = String.Empty
Public Readonly Property Address5() As String
	Get 
		Return _Address5
	End Get	
End Property	
Public Writeonly Property SetAddress5() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Address5", Value)
	End Set
End Property


Private _Postcode As String = String.Empty
Public Readonly Property Postcode() As String
	Get 
		Return _Postcode
	End Get	
End Property	
Public Writeonly Property SetPostcode() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Postcode", Value)
	End Set
End Property


Private _TelephoneNum As String = String.Empty
Public Readonly Property TelephoneNum() As String
	Get 
		Return _TelephoneNum
	End Get	
End Property	
Public Writeonly Property SetTelephoneNum() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_TelephoneNum", Value)
	End Set
End Property


Private _FaxNum As String = String.Empty
Public Readonly Property FaxNum() As String
	Get 
		Return _FaxNum
	End Get	
End Property	
Public Writeonly Property SetFaxNum() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_FaxNum", Value)
	End Set
End Property


Private _EmailAddress As String = String.Empty
Public Readonly Property EmailAddress() As String
	Get 
		Return _EmailAddress
	End Get	
End Property	
Public Writeonly Property SetEmailAddress() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_EmailAddress", Value)
	End Set
End Property


Private _Active As Boolean = False
Public Readonly Property Active() As Boolean
	Get 
		Return _Active
	End Get	
End Property	
Public Writeonly Property SetActive() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Active", Value)
	End Set
End Property


  
#End Region

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Warehouse/Warehouse.cs
using System.Collections;

namespace FSM.Entity
{
    namespace Warehouses
    {
        public class Warehouse
        {
            private DataTypeValidator _dataTypeValidator;

            public Warehouse()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            
            

            private int _WarehouseID = 0;

            public int WarehouseID
            {
                get
                {
                    return _WarehouseID;
                }
            }

            public object SetWarehouseID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_WarehouseID", value);
                }
            }

            private string _Name = string.Empty;

            public string Name
            {
                get
                {
                    return _Name;
                }
            }

            public object SetName
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Name", value);
                }
            }

            private string _Size = string.Empty;

            public string Size
            {
                get
                {
                    return _Size;
                }
            }

            public object SetSize
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Size", value);
                }
            }

            private string _Notes = string.Empty;

            public string Notes
            {
                get
                {
                    return _Notes;
                }
            }

            public object SetNotes
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Notes", value);
                }
            }

            private string _Address1 = string.Empty;

            public string Address1
            {
                get
                {
                    return _Address1;
                }
            }

            public object SetAddress1
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Address1", value);
                }
            }

            private string _Address2 = string.Empty;

            public string Address2
            {
                get
                {
                    return _Address2;
                }
            }

            public object SetAddress2
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Address2", value);
                }
            }

            private string _Address3 = string.Empty;

            public string Address3
            {
                get
                {
                    return _Address3;
                }
            }

            public object SetAddress3
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Address3", value);
                }
            }

            private string _Address4 = string.Empty;

            public string Address4
            {
                get
                {
                    return _Address4;
                }
            }

            public object SetAddress4
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Address4", value);
                }
            }

            private string _Address5 = string.Empty;

            public string Address5
            {
                get
                {
                    return _Address5;
                }
            }

            public object SetAddress5
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Address5", value);
                }
            }

            private string _Postcode = string.Empty;

            public string Postcode
            {
                get
                {
                    return _Postcode;
                }
            }

            public object SetPostcode
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Postcode", value);
                }
            }

            private string _TelephoneNum = string.Empty;

            public string TelephoneNum
            {
                get
                {
                    return _TelephoneNum;
                }
            }

            public object SetTelephoneNum
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_TelephoneNum", value);
                }
            }

            private string _FaxNum = string.Empty;

            public string FaxNum
            {
                get
                {
                    return _FaxNum;
                }
            }

            public object SetFaxNum
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_FaxNum", value);
                }
            }

            private string _EmailAddress = string.Empty;

            public string EmailAddress
            {
                get
                {
                    return _EmailAddress;
                }
            }

            public object SetEmailAddress
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_EmailAddress", value);
                }
            }

            private bool _Active = false;

            public bool Active
            {
                get
                {
                    return _Active;
                }
            }

            public object SetActive
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Active", value);
                }
            }



            
        }
    }
}